include/sapi/implementation.h

```diff
@@ -1331,36 +1325,14 @@ typedef  UINT32             TPM_CC;
 					  )
 
 #define VENDOR_COMMAND_ARRAY_SIZE   ( 0					\
-				      + CC_Vendor_TCG_Test		\
-				      )
+				      + CC_Vendor_TCG_Test)
 
 #define COMMAND_COUNT							\
     (LIBRARY_COMMAND_ARRAY_SIZE + VENDOR_COMMAND_ARRAY_SIZE)
 
 
-#ifndef MAX
-#define MAX(a, b) ((a) > (b) ? (a) : (b))
-#endif
-
-#define MAX_HASH_BLOCK_SIZE  (						\
-			      MAX(ALG_SHA1 * SHA1_BLOCK_SIZE,		\
-				  MAX(ALG_SHA256 * SHA256_BLOCK_SIZE,	\
-				      MAX(ALG_SHA384 * SHA384_BLOCK_SIZE, \
-					  MAX(ALG_SM3_256 * SM3_256_BLOCK_SIZE, \
-					      MAX(ALG_SHA512 * SHA512_BLOCK_SIZE, \
-						  0 ))))))
-
-#define MAX_DIGEST_SIZE      (						\
-			      MAX(ALG_SHA1 * SHA1_DIGEST_SIZE,		\
-				  MAX(ALG_SHA256 * SHA256_DIGEST_SIZE,	\
-				      MAX(ALG_SHA384 * SHA384_DIGEST_SIZE, \
-					  MAX(ALG_SM3_256 * SM3_256_DIGEST_SIZE, \
-					      MAX(ALG_SHA512 * SHA512_DIGEST_SIZE, \
-						  0 ))))))
-
-#if MAX_DIGEST_SIZE == 0 || MAX_HASH_BLOCK_SIZE == 0
-#error "Hash data not valid"
-#endif
+#define MAX_HASH_BLOCK_SIZE SHA512_BLOCK_SIZE
+#define MAX_DIGEST_SIZE SHA512_DIGEST_SIZE
```

Comment:
This login looks strange because 64 (SHA1_BLOCK_SIZE) will always be smaller than 128 (SHA512_BLOCK_SIZE). Since the library will support all the algs, ALG_* for all SHAs macros are set to YES this logic didn't make much sense. If we add a new algorithm this MAX_HASH_BLOCK_SIZE will have to be updated anyway.